### PR TITLE
[aws-lambda] add custom sms sender cognito trigger

### PR DESCRIPTION
**Pull request by smacker**
Reference: https://docs.aws.amazon.com/cognito/latest/developerguide/user-pool-lambda-custom-sms-sender.htm

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: [<<url here>>](https://docs.aws.amazon.com/cognito/latest/developerguide/user-pool-lambda-custom-sms-sender.htm)

### CODE REVIEW

File: types/aws-lambda/test/cognito-tests.ts [hunk 1]
```diff
@@ -271,3 +272,19 @@ const customEmailSender: CustomEmailSenderTriggerHandler = async (event, _, call
...
+    strOrNull = request.code;
+    obj = request.userAttributes;
+    objectOrUndefined = request.clientMetadata;
+
+    triggerSource === 'CustomSMSSender_AdminCreateUser';
+    triggerSource === 'CustomSMSSender_VerifyUserAttribute';
+    triggerSource === 'CustomSMSSender_UpdateUserAttribute';
+    triggerSource === 'CustomSMSSender_ResendCode';
+    triggerSource === 'CustomSMSSender_SignUp';
+    triggerSource === 'CustomSMSSender_Authentication';
```
Comment (jamesorlakin) [inline types/aws-lambda/test/cognito-tests.ts:289]: Small nit, this is missing `CustomSMSSender_ForgotPassword`

**Review by jamesorlakin — APPROVED** [review]
Thanks @smacker!

File: types/aws-lambda/test/cognito-tests.ts (hunk 1 above)
Comment (smacker) [reply types/aws-lambda/test/cognito-tests.ts:290]: thank you. Fixed.

**Review by jamesorlakin — APPROVED** [review]

**Review by andrewbranch — APPROVED** [review]

**Comment by smacker** [comment]
Ready to merge

**Pull request merged**